### PR TITLE
[new] tuxguitar derivation

### PR DESCRIPTION
**Pull request by ardumont**
- Update swt library to permit the installation for x86_64 platform
- Add tuxguitar derivation (x86, x86_64)

### CODE REVIEW

File: pkgs/applications/editors/music/tuxguitar/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{ system, stdenv, fetchurl, swt, jdk, makeWrapper, gtk, alsaLib }:
+
+let metadata = if system == "i686-linux"
+               then { arch = "x86"; sha256 = "1qmb51k0538pn7gv4nsvhfv33xik4l4af0qmpllkzrikmj8wvzlb"; }
+               else if system == "x86_64-linux"
+                    then { arch = "x86_64"; sha256 = "12af47jhlrh9aq5b3d13l7cdhlndgnfpy61gz002hajbq7i00ixh"; }
+                    else { };
```
Comment (ardumont) [inline pkgs/applications/editors/music/tuxguitar/default.nix]: I am unhappy with this. How can I do better than this?

Thanks for pointer.

**Comment by ardumont** [comment]
Also, it works!

screenshot_2015-03-07_18-00-49

**Comment by ardumont** [comment]
Ok. I see my error. I'm fixing it.

**Comment by ardumont** [comment]
The build fails because it exceeded 50 minutes.
Is there something I can do?

File: pkgs/applications/editors/music/tuxguitar/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{ system, stdenv, fetchurl, swt, jdk, makeWrapper, gtk, alsaLib }:
```
Comment (ardumont) [inline pkgs/applications/editors/music/tuxguitar/default.nix]: No longer sure the gtk deps is needed.
Will remove it to check.

File: pkgs/applications/editors/music/tuxguitar/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+in stdenv.mkDerivation rec {
+  version = "1.2";
+  name = "tuxguitar-${version}";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/tuxguitar/${name}-linux-${metadata.arch}.tar.gz";
+    sha256 = metadata.sha256;
+  };
+
+  buildInputs = [ makeWrapper ];
```
Comment (ardumont) [inline pkgs/applications/editors/music/tuxguitar/default.nix]: After a good night of sleep, I realized the build did not depend on swt, jdk and alsaLib.
Fixed it.

File: pkgs/applications/editors/music/tuxguitar/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r * $out/
+
+    wrapProgram $out/tuxguitar \
+      --prefix JAVA : "${jdk}/bin/java" \
```
Comment (ardumont) [inline pkgs/applications/editors/music/tuxguitar/default.nix]: `sh
--set JAVA "${jdk/bin/java" \
```

**Comment by ardumont** [comment]
Installed in environment.
Launched from xmonad.
(also sound ok but this could not be heard from screenshot :)

screenshot_2015-03-08_11-24-45

**Comment by ardumont** [comment]
I still don't know what to do to make the build time drop below 50 minutes.

As I see ci-travis tries to rebuild azureus (which must be dependent on the graphic library swt), I tried to build it and launch it locally to make sure this did not break.

It's ok and runs.

From nixpkgs folder:
```sh
nix-build -A azureus
./result/bin/azureus
```

screenshot_2015-03-08_11-33-13

Cheers,

**Comment by lethalman** [comment]
Looks fine thanks.

**Pull request merged**